src/Storage/TSDatabaseView.m

```diff
@@ -265,11 +266,10 @@ + (void)asyncRegisterSecondaryDevicesDatabaseView
 + (NSDate *)localTimeReceiveDateForInteraction:(TSInteraction *)interaction {
     NSDate *interactionDate = interaction.date;
 
-    if ([interaction isKindOfClass:[TSIncomingMessage class]]) {
-        TSIncomingMessage *message = (TSIncomingMessage *)interaction;
-
-        if (message.receivedAt) {
-            interactionDate = message.receivedAt;
+    if ([interaction isKindOfClass:[TSMessage class]]) {
```

Comment:
You're right, it was fine. 

For me anyway, the new implementation is clearer:
https://github.com/WhisperSystems/SignalServiceKit/pull/117/commits/99e85ecd1e48d953a7c3fff9c4afca20a9b1fada